### ISSUE TITLE
Question about generation performance and setup process

### ISSUE THREAD

**Issue by thaakeno** [mshoe/PARC#3]
Hey, first off, amazing project! I was really excited to dive into it after seeing the demos.

I finally got the generation script running on a cloud H100 GPU after working through some of the initial setup hurdles. So after many code changes, I finally generated my first motion, but it took quite a while. Just to get a benchmark, changing `num_new_motions` to 1 took the H100 around **800 seconds (~13.5 minutes)** to generate and optimize a single 3-second motion clip.

https://github.com/user-attachments/assets/b3a42a9d-b852-4258-a2ae-d3da91e341eb

This got me curious, so I looked back at the paper. In Section 7, you mention that your final model on an A6000 GPU took about **12 seconds to generate 0.5 seconds of motion**. By my math, a 3-second clip should have taken around **72 seconds** on your hardware. My H100 was more than 10x slower than that, which was surprising.

<img width="385" height="195" alt="Image" src="https://github.com/user-attachments/assets/529464ce-6f1c-4482-bfb2-5ac43226c5a3" />

So my question is: **what kind of generation times were you seeing on your end, and does the performance I'm seeing seem expected?**

For context, I had to make a few significant changes to get the project running, and I'm wondering if one of these is the source of the performance difference:

*   **PyTorch/CUDA Version:** The required `torch==1.13.1` isn't compatible with an H100. I had to upgrade to a modern PyTorch build (`cu121`) to get it to run at all. I suspect this is the main issue, and the older model architecture might not be taking advantage of the new hardware features, or maybe there's a performance regression.
*   **Setup Script Bug:** The latest version of `parc_0_setup_iter.py` had an `AttributeError` on `input_model_path.is_file()` when `input_model_path` was `None`, so I had to comment out that assertion to proceed with the initial setup.

Any insight you could offer would be awesome. I'm really looking forward to experimenting more with this.

Thanks again for the great work

**Comment by mshoe** [comment]
@Black0Moonlight Did it start producing NaN right away or did the losses blow up after some training?

**Comment by mshoe** [comment]
Ah thanks for catching that. The motions in the "initial" folder were actually using an older skeleton model where the rest-pose was an I-pose. We later updated the motions in "initial_aug" so that the rest-pose is a T-pose, but forgot to do that for the motions in "initial". So the arms are all strange looking like this:
<img width="907" height="556" alt="Image" src="https://github.com/user-attachments/assets/88239ef6-0467-4f60-b0fd-d3fca0018c91" />

I'm guessing this causes a lot more terrain penetration loss than before, and then that might mean the current initial learning rate is too high and blows up training. I just updated the initial dataset to use the current skeleton model.
